src/org/omegat/gui/exttrans/MachineTranslateTextArea.java

```diff
@@ -105,17 +115,54 @@ public MachineTranslateTextArea(IMainWindow mw) {
                 .collect(Collectors.toMap(GlossaryEntry::getSrcText, GlossaryEntry::getLocText));
     }
 
+    private void initDocument() {
+        StyleSheet baseStyleSheet = new StyleSheet();
+        HTMLEditorKit htmlEditorKit = new HTMLEditorKit();
+        baseStyleSheet.addStyleSheet(htmlEditorKit.getStyleSheet()); // Add default styles
+        Font font = getFont();
+        baseStyleSheet.addRule("body { font-family: " + font.getName() + "; "
+                + " font-size: " + font.getSize() + "; "
+                + " font-style: " + (font.getStyle() == Font.BOLD ? "bold"
+                        : font.getStyle() == Font.ITALIC ? "italic" : "normal") + "; "
+                + " color: " + Styles.EditorColor.COLOR_FOREGROUND.toHex() + "; "
+                + " background: " + Styles.EditorColor.COLOR_BACKGROUND.toHex() + ";} "
+                + ".engine {font-style: italic; text-align: right;}"
+                );
+        htmlEditorKit.setStyleSheet(baseStyleSheet);
+        setEditorKit(htmlEditorKit);
+    }
+
     /** Cycle getDisplayedTranslation **/
-    private Iterator<MachineTranslationInfo> cycle;
+    private int cycle;
 
-    public String getDisplayedTranslation() {
-        if ((cycle == null) || (!cycle.hasNext())) {
-            cycle = displayed.iterator();
-        }
-        if (!cycle.hasNext()) { // only possible if displayed.isEmpty()
+    public MachineTranslationInfo getDisplayedTranslation() {
+        if (displayed.isEmpty()) {
             return null;
         }
-        return cycle.next().result;
+        if ((cycle > displayed.size() - 1)) {
+            cycle = 0;
+        }
+        MachineTranslationInfo info = displayed.get(cycle);
+        flashSelected();
+        cycle++;
```

Comment:
I think "cycle" is a bit cryptic. Would "selectedIndex" or something be better?

Also a less verbose, common idiom for cycling an index would be

```
selectedIndex = (selectedIndex + 1) % displayed.size();
```